arviz/tests/test_data.py

```diff
@@ -413,13 +422,23 @@ def test_inference_data(self, data, eight_schools_params):
         inference_data1 = self.get_inference_data(data, eight_schools_params)
         inference_data2 = self.get_inference_data2(data, eight_schools_params)
         inference_data3 = self.get_inference_data3(data, eight_schools_params)
+        inference_data4 = self.get_inference_data4(data)
         assert hasattr(inference_data1.sample_stats, "log_likelihood")
```

Comment:
Maybe, but 4 is so low number, it is easier to see where tests fail

Reply:
Having the asserts like this will make it harder to see if asserts fail because an earlier assert failure will hide all future. A loop would be better because you could do something like
```
failed_attrs = []
for check in ((intererence_data1.prior, "theta_test"),  (intererence_data1.posterior, "theta")):
     if not hasattr(*check):
         failed_attrs.append(check)
assert failed_attrs == []
```

This will work well with pytest because then it'll show a nice diff showing what is the on the left side which failed after running all the checks

Not blocking this merge. Would be cool for a followup pr